safenode/src/storage/errors.rs

```diff
@@ -23,16 +26,47 @@ pub enum Error {
     /// Chunk not found.
     #[error("Chunk not found: {0:?}")]
     ChunkNotFound(XorName),
+    /// Register not found.
+    #[error("Register not found: {0:?}")]
+    RegisterNotFound(RegisterAddress),
+    /// NetworkData error.
+    #[error("Network data error:: {0}")]
+    NetworkData(#[from] ProtocolError),
+    /// Data authority provided is invalid.
+    #[error("Provided PublicKey could not validate signature {0:?}")]
+    InvalidSignature(PublicKey),
+    /// Bincode error.
+    #[error("Bincode error:: {0}")]
+    Bincode(#[from] bincode::Error),
```

Comment:
Mm, this as well is since old.

But I was wondering if we should maybe stick to a single serialization protocol all over? Like MsgPack. And get that in from start, to avoid building up a backlog.

Reply:
I'm going to pull this out as an issue to discuss as I think it'll need a bit of it.